src/vcf/validator.cpp

```diff
@@ -135,12 +135,19 @@ namespace ebi
                            ValidationLevel validationLevel,
                            std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs)
     {
+        boost::iostreams::filtering_istream uncompressed_input;
+        create_uncompressed_stream(input, sourceName, uncompressed_input);
+
         std::vector<char> line;
-        ebi::util::readline(input, line);
-        ebi::vcf::Version version;
-        if (ebi::vcf::is_compressed_file(sourceName, line)) {
+        ebi::util::readline(uncompressed_input, line);
+
+        std::string compression_type = ebi::vcf::get_compressed_magic_num(line);
```

Comment:
This 4 lines are checking if the file is compressed twice. it can go in another function